warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -1392,15 +1366,16 @@ protected void setupRowBasedHdfsBackedSet(String row) throws IOException {
 
         try {
             // for each of the ivarator cache dirs
-            for (IvaratorCacheDir ivaratorCacheDir : this.ivaratorCacheDirs) {
-                // get the row specific dir
-                Path rowDir = getRowDir(new Path(ivaratorCacheDir.getPathURI()), row);
-
-                FileSystem fs = ivaratorCacheDir.getFs();
-
-                // if we are not allowing reuse of directories, then delete it
-                if (!this.allowDirReuse && fs.exists(rowDir)) {
-                    fs.delete(rowDir, true);
+            if (!this.allowDirReuse) {
+                for (IvaratorCacheDir ivaratorCacheDir : this.ivaratorCacheDirs) {
+                    // get the row specific dir
+                    Path rowDir = getRowDir(new Path(ivaratorCacheDir.getPathURI()), row);
+                    FileSystem fs = ivaratorCacheDir.getFs();
+
+                    // if we are not allowing reuse of directories, then delete it
+                    if (fs.exists(rowDir)) {
+                        fs.delete(rowDir, true);
+                    }
                 }
             }
 
```

Comment:
log information on correlation between ivaratorCacheDirs and ivaratorInfo for better debugging